jsbits/diff.js

```diff
@@ -198,168 +205,152 @@ function syncChildren (os, ns, parent) {
 	    tmp = oldLastIndex - oldFirstIndex;
 	    while (tmp >= 0) {
 	      parent.removeChild(os[oldFirstIndex].domRef);
-	      os.splice(oldFirstIndex, 1);
 	      tmp--;
-	    }
-	    break;
+	    } break;
 	}
 	/* happy path, everything aligns, we continue
 	   -> oldFirstIndex -> [ a b c ] <- oldLastIndex
 	   -> newFirstIndex -> [ a b c ] <- newLastIndex
 	   check if nFirst and oFirst align, if so, check nLast and oLast
 	*/
-	else if (oFirst.key === nFirst.key) {
-	    newFirstIndex++;
-	    oldFirstIndex++;
-	    continue;
-	} else if (oLast.key === nLast.key) {
-	    newLastIndex--;
-	    oldLastIndex--;
-	    continue;
-	}
+	else if (oFirst.key === nFirst.key || oLast.key === nLast.key) {
+            while (os[oldFirstIndex].key === ns[newFirstIndex].key && os[oldLastIndex].key === ns[newLastIndex].key) {
```

Comment:
Does the order in which elements are diffed actually matter? Otherwise it seems like this loop can be removed since the two loops below cover this case.

Reply:
Yea we probably can remove this case. Doing two at once might be overrated.